### PR TITLE
History client internal types

### PR DESCRIPTION
**Pull request by vytautas-karpavicius**
**What changed?**
Switch history client to internal types. Changed immediate client usage by passing already constructed types or using thrift mappers at the call boundary. Upcoming PRs will propagate those conversions further up, until internal types reaches handlers eventually.

**Why?**
To propagate internal types throughout Cadence codebase.

**How did you test it?**
Existing tests.

**Potential risks**

### CODE REVIEW

File: service/worker/parentclosepolicy/workflow.go [hunk 1]
```diff
@@ -110,26 +110,26 @@ func ProcessorActivity(ctx context.Context, request Request) error {
 			//no-op
 			continue
 		case shared.ParentClosePolicyTerminate:
-			err = client.TerminateWorkflowExecution(nil, &h.TerminateWorkflowExecutionRequest{
+			err = client.TerminateWorkflowExecution(nil, &types.HistoryTerminateWorkflowExecutionRequest{
```
Comment (yycptt) [inline service/worker/parentclosepolicy/workflow.go:113]: Do we need to convert the error back to thrift types here for now? The error handling logic below only checks for thrift types I think.

File: service/frontend/workflowHandler.go [hunk 2]
```diff
@@ -908,24 +914,25 @@ func (wh *WorkflowHandler) RespondActivityTaskCompleted(
...
 		// result exceeds blob size limit, we would record it as failure
-		failRequest := &gen.RespondActivityTaskFailedRequest{
+		failRequest := &types.RespondActivityTaskFailedRequest{
 			TaskToken: completeRequest.TaskToken,
 			Reason:    common.StringPtr(common.FailureReasonCompleteResultExceedsLimit),
 			Details:   completeRequest.Result[0:sizeLimitError],
 			Identity:  completeRequest.Identity,
 		}
-		err = wh.GetHistoryClient().RespondActivityTaskFailed(ctx, &h.RespondActivityTaskFailedRequest{
+		err = wh.GetHistoryClient().RespondActivityTaskFailed(ctx, &types.HistoryRespondActivityTaskFailedRequest{
```
Comment (yycptt) [inline service/frontend/workflowHandler.go:923]: also add `err = thrift.FromError(err)` after this call?

File: service/frontend/workflowHandler.go [hunk 3]
```diff
@@ -1403,13 +1416,13 @@ func (wh *WorkflowHandler) RespondActivityTaskCanceledByID(
...
 		// details exceeds blob size limit, we would record it as failure
-		failRequest := &gen.RespondActivityTaskFailedRequest{
+		failRequest := &types.RespondActivityTaskFailedRequest{
 			TaskToken: token,
 			Reason:    common.StringPtr(common.FailureReasonCancelDetailsExceedsLimit),
 			Details:   cancelRequest.Details[0:sizeLimitError],
 			Identity:  cancelRequest.Identity,
 		}
-		err = wh.GetHistoryClient().RespondActivityTaskFailed(ctx, &h.RespondActivityTaskFailedRequest{
+		err = wh.GetHistoryClient().RespondActivityTaskFailed(ctx, &types.HistoryRespondActivityTaskFailedRequest{
```
Comment (yycptt) [inline service/frontend/workflowHandler.go:1425]: same here?

File: service/worker/parentclosepolicy/workflow.go (hunk 1 above)
Comment (vytautas-karpavicius) [reply service/worker/parentclosepolicy/workflow.go:114]: Yes, good catch!

File: service/frontend/workflowHandler.go (hunk 2 above)
Comment (vytautas-karpavicius) [reply service/frontend/workflowHandler.go:923]: Updated.

File: service/frontend/workflowHandler.go (hunk 3 above)
Comment (vytautas-karpavicius) [reply service/frontend/workflowHandler.go:1426]: Updated, thank you!

**Review by yycptt — APPROVED** [review]

**Pull request merged**